src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
     }
 }");
         }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationInstanceMethod1()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    void M1()
+    {
+    }
+    void M2()
+    {
+        int i=0;
+        [|M1|](i);
+    }
+}
+",
+@"
+class C
+{
+    void M1(int i)
+    {
+    }
+    void M2()
+    {
+        int i=0;
+        M1(i);
+    }
+}
+");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationInheritedMethodGetFixed()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class Base
+{
+    protected void M1()
+    {
+    }
+}
+class C1 : Base
+{
+    void M2()
+    {
+        int i = 0;
+        [|M1|](i);
+    }
+}",
+@"
+class Base
+{
+    protected void M1(int i)
+    {
+    }
+}
+class C1 : Base
+{
+    void M2()
+    {
+        int i = 0;
+        M1(i);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationInheritedMethodInMetadatGetsNotFixed()
+        {
+            await TestMissingAsync(
+    @"
+class C1
+{
+    void M2()
+    {
+        int i = 0;
+        [|GetHashCode|](i);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationLocalFunction()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    void M1()
+    {
+        int Local() => 1;
+        [|Local|](2);
+    }
+}",
+@"
+class C1
+{
+    void M1()
+    {
+        int Local(int v) => 1;
+        Local(2);
+    }
+}");
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        [Trait("TODO", "Fix broken")]
+        public async Task TestInvocationLambda1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+using System;
+class C1
+{
+    void M1()
+    {
+        Action a = () => { };
+        [|a|](2);
+    }
+}");
+            //Should be Action<int> a = (int v) => { };
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationStaticMethod()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C1
+{
+    static void M1()
+    {
+    }
+    void M2()
+    {
+        [|M1|](1);
```

Comment:
is this supposed only change decl? or all references as well? or only work if there is 1 reference?